subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultDependencySet.java

```diff
@@ -61,20 +59,16 @@ public TaskDependency getBuildDependencies() {
 
     @Override
     public boolean add(final Dependency o) {
-        warnIfConfigurationIsDeprecated();
+        assertConfigurationIsDeclarable();
         if (o instanceof AbstractModuleDependency) {
             ((AbstractModuleDependency) o).addMutationValidator(mutationValidator);
         }
         return super.add(o);
     }
 
-    private void warnIfConfigurationIsDeprecated() {
-        List<String> alternatives = ((DeprecatableConfiguration) clientConfiguration).getDeclarationAlternatives();
```

Comment:
We still support deprecating a dependency for declaration. We should keep this check around in case we called `configuration.deprecateForDecaration(<alternatives>)`. Apples to `DefaultDependencyConstraintSet` as well.